src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -712,5 +712,29 @@ private static string[] HandleClear(string[] input)
 
             return input;
         }
+
+        internal static NuGetVersion? GetSdkAnalysisLevel(IReadOnlyList<IVsTargetFrameworkInfo4> targetFrameworks)
+        {
+            string skdAnalysisLevelString = GetSingleNonEvaluatedPropertyOrNull(targetFrameworks, nameof(ProjectBuildProperties.SdkAnalysisLevel), v => v) ?? string.Empty;
+            NuGetVersion? sdkAnalysisLevel = null;
+            if (skdAnalysisLevelString != null)
+            {
+                sdkAnalysisLevel = new NuGetVersion(skdAnalysisLevelString);
+            }
+
+            return sdkAnalysisLevel;
+        }
+
+        internal static bool? GetUsingMicrosoftNETSdk(IReadOnlyList<IVsTargetFrameworkInfo4> targetFrameworks)
+        {
+            string? usingNetSdk = GetSingleNonEvaluatedPropertyOrNull(targetFrameworks, nameof(ProjectBuildProperties.UsingMicrosoftNETSdk), v => v);
+
+            if (bool.TryParse(usingNetSdk, out var result))
+            {
+                return result;
+            }
+
+            return null;
```

Comment:
This no longer compiles since you changed `UsingMicrosoftNETSdk` in `PackageSpec.cs` to be a `bool` rather than `bool?`.

Remember now that `PackageSpec.cs` defaults to `true`, that if the parsing fails, or if the string isn't set in the project, that this code needs to explicitly set the PackageSpec's property to false, do not leave the default.